### PR TITLE
WIP: Refactor handling of code blocks with just comments

### LINKED ISSUE #509: One-line node with just a comment
Input:
```php
<?php // test ?>
<?php echo 1; ?>
```
Output:
```php
<?php // test
echo 1; ?>
```

Input:
```php
<?php // test ?>
<?php echo 1; ?>
<?php echo 2; ?>
```
Output
```
[error] stdin: Error: Comment location overlaps with node location
[error]     at decorateComment (...)
```

### PR DESCRIPTION
**Pull request by czosel**
Fixes #509

### CODE REVIEW

**Comment by evilebottnawi** [comment]
@czosel i can look on this at Monday, little vacation, also we have some regression after last commit (comments), i will create issue. Also i think we should create issue in parser, maybe new option for this or modify standard behavior.

**Comment by czosel** [comment]
@evilebottnawi no rush, enjoy your vacation :wink:

File: tests/comments/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -184,15 +184,11 @@ exports[`blank_lines.php 1`] = `
 
 // Other comment
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-<?php
-// The first line of a comment
+<?php // The first line of a comment
```
Comment (czosel) [inline tests/comments/__snapshots__/jsfmt.spec.js.snap]: Related to https://github.com/glayzzle/php-parser/issues/169

File: src/printer.js [hunk 2]
```diff
@@ -1394,20 +1372,26 @@ function printLines(path, options, print, childrenAttribute = "children") {
...
       if (between && between[2]) {
-        afterOpenTag = between[2].includes("\n")
-          ? concat([
-              hardline,
-              between[2].split("\n").length > 2 ? hardline : ""
-            ])
-          : " ";
+        afterOpenTag =
+          between[2].includes("\n") &&
+          firstNestedChildNode.kind !== "lonelyComment"
```
Comment (evilebottnawi) [inline src/printer.js]: Bug with ` ` extra here

**Comment by evilebottnawi** [comment]
@czosel can you add test
```
<?php
if (true) {
    ?>inline<?php
    // comment
}
```
from https://github.com/prettier/plugin-php/issues/517

**Comment by evilebottnawi** [comment]
To be honest, I'm not sure that the idea of creating new ast nodes is good. Can you add link on issue in source code?

**Comment by evilebottnawi** [comment]
@czosel let's add test from https://github.com/prettier/plugin-php/issues/526

**Comment by evilebottnawi** [comment]
Also need fix space and we can merge

**Comment by evilebottnawi** [comment]
@czosel Please read https://github.com/prettier/plugin-php/pull/532#issue-207678433. Let's search solution without new ast nodes (fix it in parser or prepare text for parser).

**Comment by czosel** [comment]
@evilebottnawi agreed.

**Pull request closed without merging**